In [24]:
#conda install tabulate

Retrieving notices: done
Channels:
 - defaults
 - conda-forge
Platform: win-64
Solving environment: ...working... done

## Package Plan ##

  environment location: C:\Users\ikech\.conda\envs\datataa

  added / updated specs:
    - tabulate


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    tabulate-0.9.0             |  py312haa95532_0          79 KB
    ------------------------------------------------------------
                                           Total:          79 KB

The following NEW packages will be INSTALLED:

  tabulate           pkgs/main/win-64::tabulate-0.9.0-py312haa95532_0 



tabulate-0.9.0       | 79 KB     |            |   0% 
tabulate-0.9.0       | 79 KB     | ##         |  20% 
tabulate-0.9.0       | 79 KB     | ########## | 100% 
tabulate-0.9.0       | 79 KB     | ########## | 100% 
                                                     
 done
Preparing transaction

In [1]:
import pandas as pd

In [31]:
shopping_trends = pd.read_csv('shopping_trends.csv', index_col='Customer ID')
shopping_trends

Age  Gender Item Purchased     Category  Purchase Amount (USD)  \
Customer ID                                                                   
1             55    Male         Blouse     Clothing                     53   
2             19    Male        Sweater     Clothing                     64   
3             50    Male          Jeans     Clothing                     73   
4             21    Male        Sandals     Footwear                     90   
5             45    Male         Blouse     Clothing                     49   
...          ...     ...            ...          ...                    ...   
3896          40  Female         Hoodie     Clothing                     28   
3897          52  Female       Backpack  Accessories                     49   
3898          46  Female           Belt  Accessories                     33   
3899          44  Female          Shoes     Footwear                     77   
3900          52  Female        Handbag  Accessories                     81   

                  Location Size      Color  Season  Review Rating  \
Customer ID                                                         
1                 Kentucky    L       Gray  Winter            3.1   
2                    Maine    L     Maroon  Winter            3.1   
3            Massachusetts    S     Maroon  Spring            3.1   
4             Rhode Island    M     Maroon  Spring            3.5   
5                   Oregon    M  Turquoise  Spring            2.7   
...                    ...  ...        ...     ...            ...   
3896              Virginia    L  Turquoise  Summer            4.2   
3897                  Iowa    L      White  Spring            4.5   
3898            New Jersey    L      Green  Spring            2.9   
3899             Minnesota    S      Brown  Summer            3.8   
3900            California    M      Beige  Spring            3.1   

            Subscription Status Payment Method   Shipping Type  \
Customer ID                                                      
1                           Yes    Credit Card         Express   
2                           Yes  Bank Transfer         Express   
3                           Yes           Cash   Free Shipping   
4                           Yes         PayPal    Next Day Air   
5                           Yes           Cash   Free Shipping   
...                         ...            ...             ...   
3896                         No           Cash  2-Day Shipping   
3897                         No         PayPal    Store Pickup   
3898                         No    Credit Card        Standard   
3899                         No         PayPal         Express   
3900                         No  Bank Transfer    Store Pickup   

            Discount Applied Promo Code Used  Previous Purchases  \
Customer ID                                                        
1                        Yes             Yes                  14   
2                        Yes             Yes                   2   
3                        Yes             Yes                  23   
4                        Yes             Yes                  49   
5                        Yes             Yes                  31   
...                      ...             ...                 ...   
3896                      No              No                  32   
3897                      No              No                  41   
3898                      No              No                  24   
3899                      No              No                  24   
3900                      No              No                  33   

            Preferred Payment Method Frequency of Purchases  
Customer ID                                                  
1                              Venmo            Fortnightly  
2                               Cash            Fortnightly  
3                        Credit Card                 Weekly  
4                             Pay

In [32]:
shopping_trends.info()

<class 'pandas.core.frame.DataFrame'>
Index: 3900 entries, 1 to 3900
Data columns (total 18 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Age                       3900 non-null   int64  
 1   Gender                    3900 non-null   object 
 2   Item Purchased            3900 non-null   object 
 3   Category                  3900 non-null   object 
 4   Purchase Amount (USD)     3900 non-null   int64  
 5   Location                  3900 non-null   object 
 6   Size                      3900 non-null   object 
 7   Color                     3900 non-null   object 
 8   Season                    3900 non-null   object 
 9   Review Rating             3900 non-null   float64
 10  Subscription Status       3900 non-null   object 
 11  Payment Method            3900 non-null   object 
 12  Shipping Type             3900 non-null   object 
 13  Discount Applied          3900 non-null   object 
 14  Promo Code Us

In [33]:
shopping_trends.describe()

Age  Purchase Amount (USD)  Review Rating  Previous Purchases
count  3900.000000            3900.000000    3900.000000         3900.000000
mean     44.068462              59.764359       3.749949           25.351538
std      15.207589              23.685392       0.716223           14.447125
min      18.000000              20.000000       2.500000            1.000000
25%      31.000000              39.000000       3.100000           13.000000
50%      44.000000              60.000000       3.700000           25.000000
75%      57.000000              81.000000       4.400000           38.000000
max      70.000000             100.000000       5.000000           50.000000

In [34]:
description = shopping_trends.describe().to_markdown()
print(description)

|       |       Age |   Purchase Amount (USD) |   Review Rating |   Previous Purchases |
|:------|----------:|------------------------:|----------------:|---------------------:|
| count | 3900      |               3900      |     3900        |            3900      |
| mean  |   44.0685 |                 59.7644 |        3.74995  |              25.3515 |
| std   |   15.2076 |                 23.6854 |        0.716223 |              14.4471 |
| min   |   18      |                 20      |        2.5      |               1      |
| 25%   |   31      |                 39      |        3.1      |              13      |
| 50%   |   44      |                 60      |        3.7      |              25      |
| 75%   |   57      |                 81      |        4.4      |              38      |
| max   |   70      |                100      |        5        |              50      |


## Numerical Overview
* **Age:** The average age is 44 with standard deviation of 15.2 indicating a diverse age range between 18 and 70. The median age is also 44, suggesting that the mean is not significantly affected by outliers.
* **Purchase Amount:** The average purchase amount is approximately $60 with a standard deviation of 23.68.
* **Review Rating:** The reviews range from 2.5 to 5 which suggests generally positive reviews with an average of 3.75 indicating that must customers are satisfied with their purchases.
* **Previous Purchases:** The average number of previous purchases is 25.35 with a minimum of 1 purchase and a standard deviation of 14.45. 


In [35]:
gender_counts = shopping_trends['Gender'].value_counts()
gender_counts

Gender
Male      2652
Female    1248
Name: count, dtype: int64

In [38]:
frequency_counts = shopping_trends['Frequency of Purchases'].value_counts()
frequency_counts

Frequency of Purchases
Every 3 Months    1147
Every 2 Weeks     1089
Annually           572
Monthly            553
Weekly             539
Name: count, dtype: int64

In [37]:
# Combine "Bi-Weekly" and "Fortnightly" into "Every Two Weeks"
shopping_trends['Frequency of Purchases'] = shopping_trends['Frequency of Purchases'].replace(['Bi-Weekly', 'Fortnightly'], 'Every 2 Weeks')

# Combine "Every 3 Months" and "Quarterly" into "Quarterly"
shopping_trends['Frequency of Purchases'] = shopping_trends['Frequency of Purchases'].replace(['Every 3 Months', 'Quarterly'], 'Every 3 Months')

In [39]:
preferred_payment_counts = shopping_trends['Preferred Payment Method'].value_counts()
preferred_payment_counts

Preferred Payment Method
PayPal           677
Credit Card      671
Cash             670
Debit Card       636
Venmo            634
Bank Transfer    612
Name: count, dtype: int64

In [40]:
subscription_counts = shopping_trends['Subscription Status'].value_counts()
subscription_counts

Subscription Status
No     2847
Yes    1053
Name: count, dtype: int64

In [41]:
category_counts = shopping_trends['Category'].value_counts()
category_counts

Category
Clothing       1737
Accessories    1240
Footwear        599
Outerwear       324
Name: count, dtype: int64

In [42]:
combined_counts1 = pd.concat([gender_counts, frequency_counts, preferred_payment_counts, subscription_counts, category_counts], axis=0)
combined_counts1

Male              2652
Female            1248
Every 3 Months    1147
Every 2 Weeks     1089
Annually           572
Monthly            553
Weekly             539
PayPal             677
Credit Card        671
Cash               670
Debit Card         636
Venmo              634
Bank Transfer      612
No                2847
Yes               1053
Clothing          1737
Accessories       1240
Footwear           599
Outerwear          324
Name: count, dtype: int64

In [114]:
shopping_trends['Customer Segment'] = 'Other'
shopping_trends.loc[(shopping_trends['Frequency of Purchases'].isin(['Weekly', 'Every 2 Weeks'])) & (shopping_trends['Subscription Status'] == 'Yes') & (shopping_trends['Purchase Amount (USD)'] > 75), 'Customer Segment'] = 'VIP Customers'
shopping_trends.loc[(shopping_trends['Frequency of Purchases'].isin(['Weekly', 'Every 2 Weeks', 'Monthly'])) & (shopping_trends['Subscription Status'] == 'Yes') & (shopping_trends['Purchase Amount (USD)'] >= 50) & (shopping_trends['Purchase Amount (USD)'] < 75), 'Customer Segment'] = 'Loyal Customers'
shopping_trends.loc[(shopping_trends['Frequency of Purchases'].isin(['Every 3 Months', 'Annually'])) & (shopping_trends['Subscription Status'] == 'Yes'), 'Customer Segment'] = 'Inactive Subscribers'
shopping_trends.loc[(shopping_trends['Frequency of Purchases'].isin(['Every 3 Months', 'Monthly'])) & (shopping_trends['Subscription Status'] == 'No') & (shopping_trends['Purchase Amount (USD)'] >= 60) & (shopping_trends['Purchase Amount (USD)'] < 80), 'Customer Segment'] = 'Regular Customers'
shopping_trends.loc[(shopping_trends['Frequency of Purchases'].isin(['Weekly', 'Every 2 Weeks', 'Monthly'])) & (shopping_trends['Subscription Status'] == 'No') & (shopping_trends['Purchase Amount (USD)'] >= 60) & (shopping_trends['Purchase Amount (USD)'] < 80), 'Customer Segment'] = 'Potential High-Value Customers'
shopping_trends.loc[(shopping_trends['Frequency of Purchases'].isin(['Weekly', 'Every 2 Weeks', 'Monthly'])) & (shopping_trends['Subscription Status'] == 'No') & (shopping_trends['Purchase Amount (USD)'] > 80), 'Customer Segment'] = 'High-Value Non-Subscribers'
shopping_trends.loc[(shopping_trends['Frequency of Purchases'].isin(['Weekly', 'Every 2 Weeks', 'Monthly'])) & (shopping_trends['Subscription Status'] == 'No') & (shopping_trends['Purchase Amount (USD)'] < 40) & (shopping_trends['Review Rating'] > 3), 'Customer Segment'] = 'Low-spending Frequent Shoppers'
shopping_trends.loc[(shopping_trends['Frequency of Purchases'].isin(['Annually', 'Every 3 Months'])) & (shopping_trends['Subscription Status'] == 'No') & (shopping_trends['Purchase Amount (USD)'] < 40) & (shopping_trends['Review Rating'] < 3.75), 'Customer Segment'] = 'At-Risk Customers'
shopping_trends

Age  Gender Item Purchased     Category  Purchase Amount (USD)  \
Customer ID                                                                   
1             55    Male         Blouse     Clothing                     53   
2             19    Male        Sweater     Clothing                     64   
3             50    Male          Jeans     Clothing                     73   
4             21    Male        Sandals     Footwear                     90   
5             45    Male         Blouse     Clothing                     49   
...          ...     ...            ...          ...                    ...   
3896          40  Female         Hoodie     Clothing                     28   
3897          52  Female       Backpack  Accessories                     49   
3898          46  Female           Belt  Accessories                     33   
3899          44  Female          Shoes     Footwear                     77   
3900          52  Female        Handbag  Accessories                     81   

                  Location Size      Color  Season  Review Rating  \
Customer ID                                                         
1                 Kentucky    L       Gray  Winter            3.1   
2                    Maine    L     Maroon  Winter            3.1   
3            Massachusetts    S     Maroon  Spring            3.1   
4             Rhode Island    M     Maroon  Spring            3.5   
5                   Oregon    M  Turquoise  Spring            2.7   
...                    ...  ...        ...     ...            ...   
3896              Virginia    L  Turquoise  Summer            4.2   
3897                  Iowa    L      White  Spring            4.5   
3898            New Jersey    L      Green  Spring            2.9   
3899             Minnesota    S      Brown  Summer            3.8   
3900            California    M      Beige  Spring            3.1   

            Subscription Status Payment Method   Shipping Type  \
Customer ID                                                      
1                           Yes    Credit Card         Express   
2                           Yes  Bank Transfer         Express   
3                           Yes           Cash   Free Shipping   
4                           Yes         PayPal    Next Day Air   
5                           Yes           Cash   Free Shipping   
...                         ...            ...             ...   
3896                         No           Cash  2-Day Shipping   
3897                         No         PayPal    Store Pickup   
3898                         No    Credit Card        Standard   
3899                         No         PayPal         Express   
3900                         No  Bank Transfer    Store Pickup   

            Discount Applied Promo Code Used  Previous Purchases  \
Customer ID                                                        
1                        Yes             Yes                  14   
2                        Yes             Yes                   2   
3                        Yes             Yes                  23   
4                        Yes             Yes                  49   
5                        Yes             Yes                  31   
...                      ...             ...                 ...   
3896                      No              No                  32   
3897                      No              No                  41   
3898                      No              No                  24   
3899                      No              No                  24   
3900                      No              No                  33   

            Preferred Payment Method Frequency of Purchases  \
Customer ID                                                   
1                              Venmo          Every 2 Weeks   
2                               Cash          Every 2 Weeks   
3                        Credit Card                 Weekly   
4                           

In [115]:
# & (shopping_trends['Purchase Amount (USD)'] >= 50) & (shopping_trends['Purchase Amount (USD)'] < 75)
# & (shopping_trends['Review Rating'] < 3)

In [116]:
shopping_trends['Customer Segment'].value_counts()

Customer Segment
Other                             1663
Inactive Subscribers               454
High-Value Non-Subscribers         392
Potential High-Value Customers     371
Low-spending Frequent Shoppers     314
Regular Customers                  211
Loyal Customers                    186
At-Risk Customers                  174
VIP Customers                      135
Name: count, dtype: int64

In [117]:
shopping_trends[shopping_trends['Customer Segment'] == 'Other']['Frequency of Purchases'].value_counts()

Frequency of Purchases
Every 3 Months    529
Annually          351
Every 2 Weeks     349
Monthly           244
Weekly            190
Name: count, dtype: int64

In [118]:
shopping_trends[shopping_trends['Customer Segment'] == 'Other']['Subscription Status'].value_counts()

Subscription Status
No     1385
Yes     278
Name: count, dtype: int64

In [119]:
shopping_trends[shopping_trends['Customer Segment'] == 'Other']['Purchase Amount (USD)'].describe()

count    1663.000000
mean       55.474444
std        22.728776
min        20.000000
25%        40.000000
50%        50.000000
75%        78.000000
max       100.000000
Name: Purchase Amount (USD), dtype: float64

In [120]:
shopping_trends.to_csv('shopping_trends_segmented.csv', index=False)

In [11]:
shopping_trends['Payment Method'].value_counts()

Payment Method
Credit Card      696
Venmo            653
Cash             648
PayPal           638
Debit Card       633
Bank Transfer    632
Name: count, dtype: int64

In [12]:
shopping_trends['Shipping Type'].value_counts()

Shipping Type
Free Shipping     675
Standard          654
Store Pickup      650
Next Day Air      648
Express           646
2-Day Shipping    627
Name: count, dtype: int64

In [13]:
shopping_trends['Discount Applied'].value_counts()

Discount Applied
No     2223
Yes    1677
Name: count, dtype: int64

In [14]:
shopping_trends['Subscription Status'].value_counts()

Subscription Status
No     2847
Yes    1053
Name: count, dtype: int64

KeyError: "None of [Index([53, 64, 73, 90, 49, 20, 85, 34, 97, 31,\n       ...\n       81, 30, 86, 64, 78, 28, 49, 33, 77, 81],\n      dtype='int64', length=3900)] are in the [columns]"